src/components/Input/index.tsx

```diff
@@ -0,0 +1,27 @@
+import React, { FunctionComponent } from 'react';
+
+// Custom hooks
+import { useAppSelector } from '../../redux/hooks';
+
+// Styles
+import * as S from './styles';
+
+const INPUT = {
+  input: ({ type, placeholder, theme }: { type: string; placeholder: string; theme: string }) => (
+    <S.Input type={type} placeholder={placeholder} theme={theme} />
+  )
+};
+
+interface InputProps {
+  inputElement: 'input';
+  type: 'text' | 'textarea' | 'search';
```

Comment:
input type cannot have "textarea" value; you can check all available types in mdn docs and use `HTMLInputTypeAttribute` built-in type for definition